laws/src/main/scala/cats/laws/MonadErrorLaws.scala

```diff
@@ -0,0 +1,21 @@
+package cats
+package laws
+
+// Taken from http://functorial.com/psc-pages/docs/Control/Monad/Error/Class/index.html
+trait MonadErrorLaws[F[_, _], E] extends MonadLaws[F[E, ?]] {
+  implicit override def F: MonadError[F, E]
+
+  def monadErrorLeftZero[A, B](e: E, f: A => F[E, B]): IsEq[F[E, B]] =
+    F.flatMap(F.raiseError[A](e))(f) <-> F.raiseError[B](e)
+
+  def monadErrorCatch[A](e: E, f: E => F[E, A]): IsEq[F[E, A]] =
```

Comment:
Really minor, but should we call this `monadErrorHandle` since we are using "handle" instead of "catch" in other places?